File: 2021/10/part1.rb
def calculate(file_name)
  lines = File.readlines(file_name)

  score_lookup = {
    ')' => 3,
    ']' => 57,
    '}' => 1197,
    '>' => 25137
  }

  scores = lines.map do |line|
    stack = []
    score = 0
    line.chomp.chars.each do |character|
      if stack.empty? || ['(','[','{','<'].include?(character)
        stack << character
      else
        top = stack.pop
        if (top == '(' && character != ')') ||
          (top == '[' && character != ']') ||
          (top == '{' && character != '}') ||
          (top == '<' && character != '>')
          puts "ERROR - #{line.chomp} - top: #{top}, character: #{character}"
          score = score_lookup[character]
        end
      end
    end
    score
  end
  p scores.sum
end

calculate("example_input.txt")
calculate("input.txt")

File: 2022/7/part1.py

# from file_system import FileSystem, Directory, File
from utilities import parse_commands


def calculate(file_name):
  """Method"""
  with open(file_name) as file:
    lines = [line.rstrip() for line in file]

  fs = parse_commands(lines)

  fs.print()
  print(sum([s for s in fs.all_directory_sizes() if s <= 100_000]))


# calculate("example_input.txt")
calculate("input.txt")
